### DISCUSSION TITLE
Support for .mvn/jvm.config in m2e

### DISCUSSION THREAD

**Discussion by vghero** [eclipse-m2e/m2e-core#1483] [Q&A]
Is there support for ".mvn/jvm.config" in Eclipse?
I tried to put some options there but they don't seem to be picked up.
Is there a possibility to add it?

Eclipse Env:
- Maven 3.8.x (embedded)
- Eclipse 2023-03
- m2e:
![image](https://github.com/eclipse-m2e/m2e-core/assets/26437329/6c02518d-48bd-44be-b0ab-532158949d68)
-JDK 17

**Comment by laeubi** [comment]
> Is there support for ".mvn/jvm.config" in Eclipse?

There is no such support and due to the way `jvm.config` work it is currently not possible to do so, please note that `jvm.config` is a feature of the **maven launcher (!)** not maven itself, so at laest it can't work without forking a new JVM.

So main question would be what particular options you are interested in, if you want you can start eclipse itself with these options and they are picket up then by m2e.

> Is there a possibility to add it?

The only way a `jvm.config` could be taken into account would be when you launch a maven build, but I'm not sure it is implemented yet.

**Reply by mmosconi1** [reply]
> The only way a jvm.config could be taken into account would be when you launch a maven build, but I'm not sure it is implemented yet.

Meanwhile (2025-03 / m2e 2.8.0) this works for Maven builds triggered by m2e. I can't think of any other use-case where the jvm.config should be used, because it is exactly meant for that.

**Comment by vghero** [comment]
The `.mvn/jvm.config` is part of our project. We have something like this in there:

```
--add-exports jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED
--add-exports jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED
--add-opens java.base/sun.nio.ch=ALL-UNNAMED
--add-opens java.base/java.io=ALL-UNNAMED
--add-opens java.base/java.lang=ALL-UNNAMED
```
But that does not necessarily mean that I want those settings "on" all the time, but only when I'm working on that specific project which has the file put and executing code :).

As m2e takes usually care that maven projects are properly setup to run in eclipse, I thought that would include it. But as you explained it's the launcher. So that would mean m2e would have to bring in the settings from there to every execution that creates a separate jvm to apply it properly, right? Like running tests etc. I see that this will be hard to do without maybe breaking other things. But could be just an option for guys that know what it does :D.

**Reply by laeubi** [reply]
I think that m2e interprests this if you add it as compiler options, but not sure about the runtime, I think the best would be to supply an example project so one can discuss this on code and describe what works/not works and what would be the expectation on how m2e acts on those.
